btr-web/btr-common-components/components/bcros/inputs/CountriesOfCitizenship/Dropdown.vue

```diff
@@ -109,55 +110,80 @@
         </ComboboxOptions>
       </div>
     </Combobox>
+    <div v-if="hasError" class="text-sm text-red-500 pt-2">
+      {{ errors[0].message }}
+    </div>
+    <div v-if="!hasError && citizenships && citizenships.length === 0" class="text-sm pt-2 ml-2">
+      {{ $t('labels.countryOfCitizenship.hint') }}
+    </div>
   </div>
 </template>
 
 <script setup lang="ts">
 import { Combobox, ComboboxInput, ComboboxButton, ComboboxOption, ComboboxOptions } from '@headlessui/vue'
-
+import type { FormError } from '#ui/types'
 import { BtrCountryI } from '~/interfaces/btr-address-i'
 
 const emit = defineEmits<{ 'update:modelValue': [value: Array<BtrCountryI>] }>()
 const props = defineProps({
   placeholder: { type: String, default: '' },
   disabled: { type: Boolean, default: false },
-  modelValue: { type: Array<BtrCountryI>, required: true }
+  modelValue: { type: Array<BtrCountryI>, required: true },
+  errors: { type: Object as PropType<FormError[]>, required: true }
 })
 
-const nonCaCitizenships = computed({
+const citizenships = computed({
   get () {
     return props.modelValue
   },
   set (value) {
+    value.sort(sortByPriority)
     emit('update:modelValue', value)
   }
 })
 
-const allCountriesWithoutCa: Array<BtrCountryI> =
-        iscCountriesListSortedByName.filter(country => country.alpha_2 !== 'CA')
-const countriesWithoutCa = ref(allCountriesWithoutCa)
+const countryOptions = ref(citizenshipOptions)
+
+// To confirm: do we want to prioritize 'USA' in the selected citizenships
+const priorityOrder = ['CA', 'CA_PR', 'US']
```

Comment:
If you want to do the priority sort here, make sure that priorityOrder list is passed as prop.

Though it might be better, just to sort the list you want it to be sorted even before passing it to the control.

Reply:
We need to sort the list of selected options every time when it is updated. If a person already has two countries of citizenship and we add Canada, the tag 'Canada (Citizen)' would appear as the first element. 

Do you think we should pass the priorityOrder as a prop, or pass the whole comparator function as a prop?

Reply:
Just priorityOrder. 
Passing comparator function would just complicate things unnecessarily.

Btw, doesnt CA need to appear near the top at all times ?

Reply:
I just talked to Julie. We can remove this sorting logic so the chips/labels are displayed in the order as they are selected.